### ISSUE TITLE
postman collection run … –color off -e …” do still output color-codes 

### ISSUE THREAD

**Issue by atokle** [a85/POSTMan-Chrome-Extension#11891]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

I want to disable colors in output from tests run from postman cli, since I run them from Jenkins, and it’s only making a lot of noice in the log there.
Running “postman collection run -h” i get documentet that “–color off” will remove the colors

But when i try to run the command-line “postman collection run … --color off -e …” I do still get color-codes, just codes for black and grey. Here is a small example from the generated output:

^[[90m┌─────────────────────────^[[39m^[[90m┬───────────────────^[[39m^[[90m┬───────────────────┐^[[39m
^[[90m│^[[39m ^[[90m│^[[39m executed ^[[90m│^[[39m failed ^[[90m│^[[39m
^[[90m├─────────────────────────^[[39m^[[90m┼───────────────────^[[39m^[[90m┼───────────────────┤^[[39m
^[[90m│^[[39m iterations ^[[90m│^[[39m 1 ^[[90m│^[[39m 0 ^[[90m│^[[39m
^[[90m├─────────────────────────^[[39m^[[90m┼───────────────────^[[39m^[[90m┼───────────────────┤^[[39m
^[[90m│^[[39m requests ^[[90m│^[[39m 2 ^[[90m│^[[39m 0 ^[[90m│^[[39m
^[[90m├─────────────────────────^[[39m^[[90m┼───────────────────^[[39m^[[90m┼───────────────────┤^[[39m
^[[90m│^[[39m test-scripts ^[[90m│^[[39m 4 ^[[90m│^[[39m 0 ^[[90m│^[[39m
^[[90m├─────────────────────────^[[39m^[[90m┼───────────────────^[[39m^[[90m┼───────────────────┤^[[39m
^[[90m│^[[39m prerequest-scripts ^[[90m│^[[39m 2 ^[[90m│^[[39m 0 ^[[90m│^[[39m

### Steps To Reproduce

1. run the command-line “postman collection run … --color off -e …” and redirect output to a file
2. Open the file with the output and check the table in the output
3. The table should not have color-formatting codes. But have code ^[[90m and ^[[39m that make it look like it have no colors even though the code still are there

### Operating System

Linux

### Postman Version

10.12.0

### Postman Platform

Postman App

### User Account Type

Signed Out User

### Additional Context?

I did choose "Postman App" as platform. But is using the postman cli.

**Comment by malvikach** [comment]
@atokle I can see this behavior too. This might not be a bug since this is the expected behavior on the terminal. I can check and confirm this further. 

Meanwhile, it will help me to know a bit more about your use case- why are you writing the output in the file and how do you plan to use it? 

Since you are using Postman CLI to run the collections, the results are also available in Postman (you can click the link that is available at the end of the run).

**Comment by atokle** [comment]
@malvikach I'm not sure that this should be the expected behavior. Usually when turning color off, it's not because you don't like flashy colors, but because the environment it's run in is not able to show the codes.

My use-case is that I run it on Jenkins build-server, and in the console log the codes will not be interpreted into colors and is just making the log harder to read.

The reason I suggested to output it to a file was just that you should not be fooled because it was not shown if you run the command in a terminal that supported the codes.

**Comment by malvikach** [comment]
@atokle Thanks for clarifying. I can see the problem on the Jenkins server when viewing the logs as console output or plaintext. 
<img width="1033" alt="Screenshot 2023-04-04 at 6 27 21 PM" src="https://user-images.githubusercontent.com/89449769/229799076-bdca48ed-03a7-4214-a9da-ce2552d0826c.png">

Agreed that this is difficult to consume and doesn't make sense. We can look into how this can be addressed.

Meanwhile, if you are using BlueOcean on Jenkins, you can see the console output formatted in a much better GUI as shown below:

<img width="1390" alt="Screenshot 2023-04-04 at 6 27 31 PM" src="https://user-images.githubusercontent.com/89449769/229799678-7ad0976a-3126-4ddf-a8bb-90c25a2b76cd.png">

Let me know if the above alternative is helpful or something you use.

Regards,
Malvika

**Comment by atokle** [comment]
Thank you. I must check if we have or can install BlueOcean. The administrators on our Jenkins-server is very restrictive about plugins, but if I get that it would solve my case

**Comment by robertmryan** [comment]
The question is not how it appears in the terminal window, but a few escape sequences are still slipping through.

For example, in Postman CLI 1.1.1 on macOS, with `--color on`, I see:


![](https://global.discourse-cdn.com/business6/uploads/getpostman/original/2X/b/bc0268fe69d4a3b2bbf736dd6c5cb300cc7fa6b5.png)

But with `--color off`:

![](https://global.discourse-cdn.com/business6/uploads/getpostman/original/2X/2/255c0cc4a904d89b0b8e5fa832ccdd1726a17c44.png)

At first glance, it looks like the final “failure”/“detail” string is the only coloring that is still being output. But if you open the output in an editor, you can see that the ascii-art-table still has all the escape sequences in it, too. E.g., having piped the `--color off` rendition to a file, when I open that in my editor, I can see all of the escape sequences that are still there:

![](https://global.discourse-cdn.com/business6/uploads/getpostman/original/2X/d/d522daf74733f8060d088e2a4278ed93f920906f.png)

Bottom line, `--color off` appears to be omitting some escape sequences, but not all.
For now, I do a regex replacement of `[\x1b]\[\d+m` with an empty string and it gets rid of these extraneous escape sequences, but it is annoying to have any escape sequences in the output when we do `--color off`.

[migrated from https://community.postman.com/t/postman-collection-run-color-off-e-do-still-output-color-codes/42700/8?u=robert.m.ryan]

**Comment by robertmryan** [comment]
FWIW, in case it helps anyone else, I pipe my output through `sed`:

```sh
postman collection run collection.json --color off | sed -E 's/\x1b\[[0-9]+m//g' > output.txt
```

**Comment by aniketbansal-postman** [comment]
Hi @atokle , @robertmryan , please upgrade to v1.31.3 where this issue is fixed
